### DISCUSSION TITLE
Caching routes on Forge (with Octane) => PHP Fatal error:  Allowed memory size exceeded

### DISCUSSION THREAD

**Discussion by LTKort** [livewire/livewire#5825] [1. Help]
When I try to deploy my site on Forge (using Octane), and call a route:cache, I get the following error:

```PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /home/forge/xxx/vendor/laravel/serializable-closure/src/Serializers/Native.php on line 222```

```PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /home/forge/xxx/vendor/symfony/error-handler/Error/FatalError.php on line 21```

It works, as soon as I exclude the Livewire routes in the command (\Illuminate\Foundation\Console\RouteCacheCommand), using; 
```php
foreach ($routes as $route) {
  if (\Illuminate\Support\Str::contains($route->uri, 'livewire')) {
       continue;
   }
   $route->prepareForSerialization();
}
.... 
```

But then the clear command gives; `Call to undefined method Closure::__set_state()` 

Locally the caching works, but with Forge & Octane it seems to give an error.

Is this a problem with Octane? Or is it something else?

**Comment by enkrs** [comment]
I think it's not Forge or Octane specific. I think it's something with Livewire v3 routes. I can reproduce it with vanilla PHP locally on empty Laravel install:

    > composer create-project laravel/laravel livewireroutes
    > cd livewireroutes
    > php artisan route:cache

       INFO  Routes cached successfully.  

    > composer require livewire/livewire:^3.0@beta 
    > php artisan route:cache

    PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /home/enkrs/livewireroutes/vendor/laravel/serializable-closure/src/Serializers/Native.php on line 222
    PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 65536 bytes) in /home/enkrs/livewireroutes/vendor/composer/ClassLoader.php on line 582

If I increase the PHP memory limit to something ridiculous, the command still does not succeed, instead it segfaults.

    php -d memory_limit=2048M artisan route:cache
    segmentation fault (core dumped)  php -d memory_limit=2048M artisan route:cache

The error only occurs if we try to compile any of these two routes - `livewire/update` and `livewire/livewire.js`. Updated \Illuminate\Foundation\Console\RouteCacheCommand for testing:

```php
         foreach ($routes as $route) {     
              if ($route->uri === 'livewire/update'     
                  or $route->uri === 'livewire/livewire.js'
              ) {                                                                 
                  continue;                                                                                     
              }                                                                                                 
              $route->prepareForSerialization();                                                                
          }
```

**Reply by nuernbergerA** [reply]
I will have a look

**Reply by nuernbergerA** [reply]
opened a PR with a failing test https://github.com/livewire/livewire/pull/5910
